src/screen.c

```diff
@@ -4914,6 +4917,47 @@ set_chars_option(win_T *wp, char_u *value, int is_listchars, int apply,
 		    break;
 		}
 
+		if (!is_listchars && STRCMP(tab[i].name.string, "ellipsis") == 0)
+		{
+		    if (round == 0)
+		    {
+			ellipsis_len = 0;
+			while (*s != NUL && *s != ',')
+			{
+			    c1 = get_encoded_char_adv(&s);
+			    if (!vim_isprintc(c1))
+				return field_value_err(errbuf, errbuflen,
+				     e_wrong_character_width_for_field_str,
+				     tab[i].name.string);
+			    ++ellipsis_len;
+			}
+			if (ellipsis_len == 0)
+			    return field_value_err(errbuf, errbuflen,
+				e_wrong_number_of_characters_for_field_str,
+				tab[i].name.string);
+		    }
+		    else
+		    {
+			vim_free(fill_chars.ellipsis);
+			fill_chars.ellipsis = ALLOC_MULT(int, ellipsis_len + 1);
+
+			if (fill_chars.ellipsis != NULL)
+			{
+			    int pos = 0;
+			    while (*s != NUL && *s != ',' && pos < ellipsis_len)
```

Comment:
`pos < ellipsis_len` should not be in the loop condition, otherwise it'll lead to an error when specifying `ellipsis:` in `'fillchars'` twice and the second `ellipsis:` is longer than the first.

Reply:
I'll push a commit to fix this, and add a bit more tests for option values and avoid unnecessary allocations.